### PR TITLE
feat: Extending the UAG Lite to the UAG Pro.

### PR DESCRIPTION
**Pull request by RajkiranBagal**
### Description
 - Extending the UAG Lite to UAG Pro, All needed functions.

### Types of changes
New feature (non-breaking change which adds functionality)

### How has this been tested?
 - Test Cases.
 
### Checklist:
- [x] My code is tested
- [ ] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: admin-core/ajax/common-settings.php [hunk 1]
```diff
@@ -677,6 +680,90 @@ public function copy_paste() {
 		wp_send_json_success( $response_data );
 
 	}
+	/**
+	 * Save settings.
```
Comment (AbhijitNage123) [inline admin-core/ajax/common-settings.php:684]: @since tag missing.

File: admin-core/ajax/common-settings.php [hunk 2]
```diff
@@ -677,6 +680,90 @@ public function copy_paste() {
...
+		\UAGB_Admin_Helper::update_admin_settings_option( 'uag_social', $social );
+
+		$response_data = array(
+			'messsage' => __( 'Successfully saved data!', 'ultimate-addons-for-gutenberg' ),
+		);
+		wp_send_json_success( $response_data );
+	}
+	/**
+	 * Save settings.
+	 *
```
Comment (AbhijitNage123) [inline admin-core/ajax/common-settings.php:739]: @Since tag missing

File: admin-core/ajax/common-settings.php [hunk 3]
```diff
@@ -779,6 +866,41 @@ public function enable_block_responsive() {
 		wp_send_json_success( $response_data );
 
 	}
+	/**
+	 * Save settings.
+	 *
```
Comment (AbhijitNage123) [inline admin-core/ajax/common-settings.php:871]: @Since tag missing

File: classes/class-uagb-block.php [hunk 4]
```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * UAGB Block.
+ *
+ * @since 2.0.0
```
Comment (AbhijitNage123) [inline classes/class-uagb-block.php:5]: It should be @Since 2.1.0

**Review by supriyajjadhav — APPROVED** [review]

File: includes/blocks/tabs/block.php [hunk 5]
```diff
@@ -9,15 +9,25 @@
...
+	'description'         => __( 'Display your content under different tabs.', 'ultimate-addons-for-gutenberg' ),
+	'default'             => true,
+	'extension'           => false,
+	'js_assets'           => array( 'uagb-tabs-js' ),
+	'priority'            => Spectra_Block_Prioritization::get_block_priority( 'tabs' ),
+	'deprecated'          => false,
+	'static_dependencies' => array(
+		'uagb-tabs-js' => array(
+			'src'  => UAGB_Scripts_Utils::get_js_url( 'tabs' ),
+			'dep'  => array( 'jquery' ),
```
Comment (AbhijitNage123) [inline includes/blocks/tabs/block.php:26]: Remove Jquery dependency

File: includes/blocks/table-of-contents/block.php [hunk 6]
```diff
@@ -9,15 +9,25 @@
...
+	'description'         => __( 'Add a table of contents to allow page navigation.', 'ultimate-addons-for-gutenberg' ),
+	'default'             => true,
+	'extension'           => false,
+	'js_assets'           => array( 'uagb-table-of-contents' ),
+	'priority'            => Spectra_Block_Prioritization::get_block_priority( 'table-of-contents' ),
+	'deprecated'          => false,
+	'static_dependencies' => array(
+		'uagb-table-of-contents' => array(
+			'src'  => UAGB_Scripts_Utils::get_js_url( 'table-of-contents' ),
+			'dep'  => array( 'jquery' ),
```
Comment (AbhijitNage123) [inline includes/blocks/table-of-contents/block.php:26]: Remove Jquery dependency

File: includes/blocks/star-rating/block.php [hunk 7]
```diff
@@ -16,7 +16,10 @@
...
 	'description'      => __( 'Display customizable star ratings on your page.', 'ultimate-addons-for-gutenberg' ),
 	'default'          => true,
-	'js_assets'        => array(),
-	'priority'         => 26,
+	'priority'         => Spectra_Block_Prioritization::get_block_priority( 'star-rating' ),
 	'deprecated'       => false,
+	'dynamic_assets'   => array(
+		'dir' => 'star-rating',
+	),
+	'js_assets'        => array(),
```
Comment (AbhijitNage123) [inline includes/blocks/star-rating/block.php:24]: Is it neccesary to add js_assets props if assets are not present for any block?

File: includes/blocks/lottie/block.php [hunk 8]
```diff
@@ -9,14 +9,28 @@
...
+	'title'               => __( 'Lottie Animation', 'ultimate-addons-for-gutenberg' ),
+	'description'         => __( 'Add customizable lottie animation on your page.', 'ultimate-addons-for-gutenberg' ),
+	'default'             => true,
+	'js_assets'           => array( 'uagb-lottie-js', 'uagb-bodymovin-js' ),
+	'priority'            => Spectra_Block_Prioritization::get_block_priority( 'lottie' ),
+	'deprecated'          => false,
+	'static_dependencies' => array(
+		'uagb-lottie-js'    => array(
+			'src'        => UAGB_Scripts_Utils::get_js_url( 'lottie' ),
+			'dep'        => array( 'jquery', 'uagb-bodymovin-js' ),
```
Comment (AbhijitNage123) [inline includes/blocks/lottie/block.php:25]: Remove Jquery dep

File: includes/blocks/inline-notice/block.php [hunk 9]
```diff
@@ -9,15 +9,29 @@
...
+	'description'         => __( 'Highlight important information using inline notice block.', 'ultimate-addons-for-gutenberg' ),
+	'default'             => true,
+	'extension'           => false,
+	'js_assets'           => array( 'uagb-inline-notice-js', 'uagb-cookie-lib' ),
+	'priority'            => Spectra_Block_Prioritization::get_block_priority( 'inline-notice' ),
+	'deprecated'          => false,
+	'static_dependencies' => array(
+		'uagb-inline-notice-js' => array(
+			'src'        => UAGB_Scripts_Utils::get_js_url( 'inline-notice' ),
+			'dep'        => array( 'jquery', 'uagb-cookie-lib' ),
```
Comment (AbhijitNage123) [inline includes/blocks/inline-notice/block.php:26]: Remove Jquery dep

File: includes/blocks/how-to-step/block.php [hunk 10]
```diff
@@ -9,10 +9,13 @@
...
-	'js_assets'   => array(),
-	'deprecated'  => false,
+	'slug'           => '',
+	'title'          => __( 'Step', 'ultimate-addons-for-gutenberg' ),
+	'description'    => __( 'Add relevant content for this step.', 'ultimate-addons-for-gutenberg' ),
+	'default'        => true,
+	'dynamic_assets' => array(
+		'dir' => 'how-to-step',
+	),
+	'js_assets'      => array(),
```
Comment (AbhijitNage123) [inline includes/blocks/how-to-step/block.php:19]: If there are not js for the block is it necessary to add js_assets

File: includes/blocks/forms/block.php [hunk 11]
```diff
@@ -9,14 +9,24 @@
...
+	'title'               => __( 'Form', 'ultimate-addons-for-gutenberg' ),
+	'description'         => __( 'Add easily customizable forms to gather information.', 'ultimate-addons-for-gutenberg' ),
+	'default'             => true,
+	'js_assets'           => array( 'uagb-forms-js' ),
+	'priority'            => Spectra_Block_Prioritization::get_block_priority( 'forms' ),
+	'deprecated'          => false,
+	'static_dependencies' => array(
+		'uagb-forms-js' => array(
+			'src'  => UAGB_Scripts_Utils::get_js_url( 'forms' ),
+			'dep'  => array( 'jquery' ),
```
Comment (AbhijitNage123) [inline includes/blocks/forms/block.php:25]: Remove Jquery dep

**Review by AbhijitNage123 — CHANGES_REQUESTED** [review]

**Comment by AbhijitNage123** [comment]
@bsf-zanev Please run grunt cmd `grunt bump-version --ver=2.1.0` it replce all x.x.x to 2.1.0

File: classes/class-uagb-block.php (hunk 4 above)
Comment (bsf-zanev) [reply classes/class-uagb-block.php:5]: Updated.

File: admin-core/ajax/common-settings.php (hunk 3 above)
Comment (bsf-zanev) [reply admin-core/ajax/common-settings.php:871]: Updated.

File: admin-core/ajax/common-settings.php (hunk 2 above)
Comment (bsf-zanev) [reply admin-core/ajax/common-settings.php:739]: Updated.

File: admin-core/ajax/common-settings.php (hunk 1 above)
Comment (bsf-zanev) [reply admin-core/ajax/common-settings.php:684]: Updated

File: includes/blocks/how-to-step/block.php (hunk 10 above)
Comment (bsf-zanev) [reply includes/blocks/how-to-step/block.php:19]: All 'js_assets' keys have been safely removed.

File: includes/blocks/star-rating/block.php (hunk 7 above)
Comment (bsf-zanev) [reply includes/blocks/star-rating/block.php:24]: All 'js_assets' keys have been safely removed.

File: includes/blocks/forms/block.php (hunk 11 above)
Comment (bsf-zanev) [reply includes/blocks/forms/block.php:25]: Safely removed unrequired jQuery dependency and tested.

File: includes/blocks/inline-notice/block.php (hunk 9 above)
Comment (bsf-zanev) [reply includes/blocks/inline-notice/block.php:26]: Safely removed unrequired jQuery dependency and tested.

File: includes/blocks/lottie/block.php (hunk 8 above)
Comment (bsf-zanev) [reply includes/blocks/lottie/block.php:25]: Safely removed unrequired jQuery dependency and tested.

File: includes/blocks/table-of-contents/block.php (hunk 6 above)
Comment (bsf-zanev) [reply includes/blocks/table-of-contents/block.php:26]: Safely removed unrequired jQuery dependency and tested.

File: includes/blocks/tabs/block.php (hunk 5 above)
Comment (bsf-zanev) [reply includes/blocks/tabs/block.php:26]: Safely removed unrequired jQuery dependency and tested.

File: includes/blocks/post-timeline/block.php [hunk 12]
```diff
@@ -9,15 +9,25 @@
...
+	'title'               => __( 'Post Timeline', 'ultimate-addons-for-gutenberg' ),
+	'description'         => __( 'Create an attractive timeline to display your posts.', 'ultimate-addons-for-gutenberg' ),
+	'default'             => true,
+	'extension'           => false,
+	'priority'            => Spectra_Block_Prioritization::get_block_priority( 'post-timeline' ),
+	'deprecated'          => false,
+	'static_dependencies' => array(
+		'uagb-timeline-js' => array(
+			'src'  => UAGB_Scripts_Utils::get_js_url( 'timeline' ),
+			'dep'  => array( 'jquery' ),
```
Comment (AbhijitNage123) [inline includes/blocks/post-timeline/block.php:25]: Remove JQuery dep

**Review by AbhijitNage123 — CHANGES_REQUESTED** [review]

**Comment by AbhijitNage123** [comment]
Please add a `= 2.1.0 - TUESDAY, 29th NOVEMBER 2022 =` line in changelog of readme  and run `grunt readme` so README.md also get updated

File: includes/blocks/post-timeline/block.php (hunk 12 above)
Comment (bsf-zanev) [reply includes/blocks/post-timeline/block.php:25]: Updated.

**Comment by bsf-zanev** [comment]
> Please add a `= 2.1.0 - TUESDAY, 29th NOVEMBER 2022 =` line in changelog of readme and run `grunt readme` so README.md also get updated

This will be updated during the NR-2.1 Merge.

**Review by AbhijitNage123 — APPROVED** [review]

**Review by supriyajjadhav — APPROVED** [review]

**Pull request merged**